test/e2e/e2e_test.go

```diff
@@ -703,12 +702,22 @@ func checkUsers(env *completedEnvironment) {
 }
 
 // if the doer returns true, consider it done, and stop retrying
-func retryFor(d time.Duration, doer func() bool) {
```

Comment:
This was changed to a timer/ticker approach, which seems more golang-friendly, and it now fails directly if the timeout is met, which wasn't previously unless the desired state was checked after this method call (which wasn't in some cases).